### PR TITLE
Added a New Combination_Sum_II in Backtracking.

### LINKED ISSUE #9908: [ADD]: Combination_Sum II
### Feature description

The count-and-say sequence is a series of number strings that follows a recursive formula. The first number in the sequence is "1". 

To get the next number in the sequence, you "say" the previous number and then convert it into a new number string.

 To "say" a number string, you need to split it into the smallest number of substrings possible, where each substring has only one unique digit. For each of these substrings, you say the number of digits in it and then the digit itself. 

Finally, you concatenate all the said digits. For instance, if the number string is "3322251", here's how you would "say" it and convert it.

Web capture_6-10-2023_15625_leetcode com

### PR DESCRIPTION
**Pull request by imSanko**
### Describe your change:

Added a New Combination_Sum_II in Backtracking.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #9908".

### CODE REVIEW

**Comment by imSanko** [comment]
@cclauss can you check the PR

File: backtracking/combination_sum_II.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+Constraints:
+The input n is a positive integer.
+The input n is bounded by 1 <= n <= 30.
+This constraint is typically imposed to limit the size of the output string,
+as the "Count and Say" sequence grows exponentially with n.
+"""
+
+class Solution:
+    def count_and_say(self, n: int) -> str:
```
[the remark this answers is not in the document]
Comment (imSanko) [reply backtracking/combination_sum_II.py:16]: class Solution:
    def count_and_say(self, term_number: int) -> str:
        """
        Generate the nth term of the "Count and Say" sequence.

        Args:
            term_number (int): The term number to generate.

        Returns:
            str: The nth term of the "Count and Say" sequence.

        Example:
            >>> solution = Solution()
            >>> solution.count_and_say(1)
            '1'
            >>> solution.count_and_say(4)
            '1211'
        """
        # Base case
        if term_number == 1:
            return "1"

        # Recursive case
        prev_term = self.count_and_say(term_number - 1)
        result = ""
        count = 1

        # Iterate through the previous term and generate the current term
        for i in range(len(prev_term)):
            if i < len(prev_term) - 1 and prev_term[i] == prev_term[i + 1]:
                count += 1
            else:
                result += str(count) + prev_term[i]
                count = 1

        return result

# Run doctests
if __name__ == "__main__":
    import doctest
    doctest.testmod()

File: backtracking/combination_sum_II.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+Constraints:
+The input n is a positive integer.
+The input n is bounded by 1 <= n <= 30.
+This constraint is typically imposed to limit the size of the output string,
+as the "Count and Say" sequence grows exponentially with n.
+"""
+
+
+class Solution:
```
Comment (cclauss) [inline backtracking/combination_sum_II.py:12]: A class named `Solution` is almost always a bad idea especially when it has no variables.  You could lose the `class` and just have a function called `count_and_say()` but that function name has nothing to do with `combination_sum` which is the filename.  If we already have a `combination_sum` then I doubt we need a second one.  Closing...  We have more than 100 open pull requests.

File: bit_manipulation/tempCodeRunnerFile.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Information on 2's complement: https://en.wikipedia.org/wiki/Two%27s_complement
```
Comment (cclauss) [inline bit_manipulation/tempCodeRunnerFile.py:1]: # Bad file name.

**Comment by cclauss** [comment]
All three checks fail.

**Pull request closed without merging**